docs/guides/table_migration.md

```diff
@@ -0,0 +1,159 @@
+# Table migration
+
+SQLMesh projects can read directly from tables not managed by SQLMesh, but in some scenarios it may be useful to migrate an existing table into a SQLMesh project.
+
+This guide describes two methods for migrating existing tables into a SQLMesh project.
+
+## Do you need to migrate?
+
+SQLMesh does not assume it manages all data sources: SQL models can read from any data source accessible by the SQL engine, treating them as [external models](../concepts/models/model_kinds.md#external) that include column-level lineage or as generic sources. This approach is preferred to migrating existing tables into a SQLMesh project.
+
+You should only migrate a table if both of the following are true:
+1. The table's upstream source is active (i.e., the table is ingesting from a source that will continue generating new data)
```

Comment:
I think "upstream source is active" is confusing and it requires what is in parentheses. Maybe just remove it and say what is in parentheses directly?